hax-lib-protocol/src/cal.rs

```diff
@@ -0,0 +1,194 @@
+//! This module defines a cryptographic abstraction layer for use in
+//! hax protocol specifications.
+
+use crate::ProtocolError;
+
+/// An abstract Diffie-Hellman scalar.
+#[derive(Clone)]
+pub struct DHScalar(Vec<u8>);
+
+impl DHScalar {
+    /// Wrap bytes into a Diffie-Hellman scalar. Does *not* perform
+    /// input validation.
+    pub fn from_bytes(bytes: &[u8]) -> Self {
```

Comment:
Is there a reason not to use `From`? This is fine too. Just wondering.

Reply:
No reason, I didn't think of implementing `From`, when adapting the `hacspec_lib` API to this.